docs/source/cpp_api/clustering.rst

```diff
@@ -0,0 +1,25 @@
+Clustering
+==========
+
+This page provides C++ class references for the publicly-exposed elements of the clustering package.
+
+K-Means
+#######
+
+.. doxygennamespace:: raft::cluster
```

Comment:
Yep, I think that's a good plan indeed. For 22.04 I think we should keep it the way it is just to avoid any potential breaking changes right before burndown.